typeclasses/src/main/scala/Ordering.scala

```diff
@@ -0,0 +1,55 @@
+trait Ordering[A]:
+  extension (a: A) def compare(b: A): Order
```

Comment:
That's a little weird. Usually, you'd define `compare(lhs: A, rhs: A)`, and then maybe add extension methods for `>`, `<`, ...